contributors/design-proposals/storage/csi-migration.md

```diff
@@ -236,34 +237,49 @@ with the in-tree plugin, the VolumeAttachment object becomes orphaned.
 
 ### In-line Volumes
 
-In-line controller calls are a special case because there is no PV. In this case
-we will translate the volume source and copy it to the field
-VolumeAttachment.Spec.Source.VolumeAttachmentSource.InlineVolumeSource. The
-VolumeAttachment name must be made with the CSI Translated version of the
+In-line controller calls are a special case because there is no PV. In this case,
+we will translate the in-line Volume into a PersistentVolumeSpec using
+plugin-specific translation logic in the CSI translation library method,
+`TranslateInTreeInlineVolumeToPVSpec`. The resulting PersistentVolumeSpec will
+be stored in a new field `VolumeAttachment.Spec.Source.VolumeAttachmentSource.InlineVolumeSpec`.
+
+The plugin-specific CSI translation logic invoked by `TranslateInTreeInlineVolumeToPVSpec`
```

Comment:
Can you add a note here, that the PV spec that the translation library produces must be able to pass PV API validation.

Reply:
And also a note that we need to run the InlineVolumeSpec through the same defaulting and validation functions that we do for PV

Reply:
> And also a note that we need to run the InlineVolumeSpec through the same defaulting and validation functions that we do for PV

I think that would happen automatically (but double check the generated `SetDefaults_VolumeAttachment` method) if `SetDefaults_PersistentVolume` was split out to set defaults relevant to all consumers of `PersistentVolumeSpec` (like `VolumeMode`) in a `SetDefaults_PersistentVolumeSpec` function:
```
func SetDefaults_PersistentVolume(obj *v1.PersistentVolume) {
	if obj.Status.Phase == "" {
		obj.Status.Phase = v1.VolumePending
	}
	if obj.Spec.PersistentVolumeReclaimPolicy == "" {
		obj.Spec.PersistentVolumeReclaimPolicy = v1.PersistentVolumeReclaimRetain
	}
	if obj.Spec.VolumeMode == nil {
		obj.Spec.VolumeMode = new(v1.PersistentVolumeMode)
		*obj.Spec.VolumeMode = v1.PersistentVolumeFilesystem
	}
}
```

Reply:
> Can you add a note here, that the PV spec that the translation library produces must be able to pass PV API validation.

You can also add tests for this in kubernetes/kubernetes, which has visibility to a) the API types, b) the translation library, and c) the API validation functions